Failed to build on x86_64-darwin, shall we set platforms to linux only?

```
FAILED: CMakeFiles/diff_match_patch.dir/third_party/diff_match_patch/diff_match_patch.cpp.o
/nix/store/vn148wjd5f92yjbmlbz0rbi195daqc8a-clang-wrapper-7.1.0/bin/clang++ -DQT_CORE_LIB -DQT_NO_DEBUG -I/tmp/nix-build-cpeditor-6.9.3.drv-0/source/build -I/tmp/nix-build-cpeditor-6.9.3.drv-0/source -I/tmp/nix-build-cpeditor-6.9.3.drv-0/source/build/diff_match_patch_autogen/include -I/tmp/nix-build-cpeditor-6.9.3.drv-0/source/generated -I/tmp/nix-build-cpeditor-6.9.3.drv-0/source/src -isystem /nix/store/8mkvcnwm55ahy2lra5krhc1zxbicx6wx-qtbase-5.12.10-dev/include/QtCore -isystem /nix/store/8mkvcnwm55ahy2lra5krhc1zxbicx6wx-qtbase-5.12.10-dev/./mkspecs/macx-clang -O3 -DNDEBUG -mmacosx-version-min=10.14 -fPIC -std=gnu++14 -MD -MT CMakeFiles/diff_match_patch.dir/third_party/diff_match_patch/diff_match_patch.cpp.o -MF CMakeFiles/diff_match_patch.dir/third_party/diff_match_patch/diff_match_patch.cpp.o.d -o CMakeFiles/diff_match_patch.dir/third_party/diff_match_patch/diff_match_patch.cpp.o -c /tmp/nix-build-cpeditor-6.9.3.drv-0/source/third_party/diff_match_patch/diff_match_patch.cpp
/tmp/nix-build-cpeditor-6.9.3.drv-0/source/third_party/diff_match_patch/diff_match_patch.cpp:2038:47: error: no member named 'SkipEmptyParts' in namespace 'Qt'
  QStringList text = textline.split("\n", Qt::SkipEmptyParts);
                                          ~~~~^
1 error generated.
```

--- a/pkgs/applications/editors/cpeditor/default.nix
+++ b/pkgs/applications/editors/cpeditor/default.nix
@@ -1,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, qtbase
+, qttools
+, wrapQtAppsHook
+, cmake
+, ninja
+, python3
+, runtimeShell
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cpeditor";
+  version = "6.10.1";
+
+  src = fetchFromGitHub {
+    owner = "cpeditor";
+    repo = "cpeditor";
+    rev = version;
+    sha256 = "sha256-SIREoOapaZTLtqi0Z07lKmNqF9a9qIpgGxuhqaY3yfU=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake ninja pkg-config wrapQtAppsHook python3 ];
+  buildInputs = [ qtbase qttools ];
+
+  postPatch = ''
+    substituteInPlace src/Core/Runner.cpp --replace "/bin/bash" "${runtimeShell}"
+  '';
+
+  meta = with lib; {
+    description = "An IDE specially designed for competitive programming";
+    homepage = "https://cpeditor.org";
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ rewine ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -24505,6 +24505,8 @@ with pkgs;
     ffmpeg = ffmpeg-full;
   };
 
+  cpeditor = libsForQt515.callPackage ../applications/editors/cpeditor { };
+
   csa = callPackage ../applications/audio/csa { };
 
   csound = callPackage ../applications/audio/csound { };

